More text on ACK rate in the applicability draft (#386)
quicwg/ops-drafts: draft-ietf-quic-applicability.md

# ACK-only packets on constrained links

The stragegy on sending acknowlegdements in QUIC version 1 is adopted from
TCP, meaning that usually an QUIC implementation is expected to send one
packet containing an ACK frame for each two packets received.
The cost of sending acknowledgments - in processing cost or link
utilization - could be a significant proportion of available resources if
these resources are constrained.  "ACK-only" packets are not reliably
detectable in QUIC, so network ACK thinning done for TCP is not suitable

> for QUIC traffic. Instead QUIC endpoint need to control the ACK rate

Reviewer: /instead QUIC endpoint need/instead a QUIC endpoint needs/

> appropriately and some implenentation may even provide an interface to

Reviewer: /some implementation may even provide an interface to impact/implementations provide an interface that can reduce/ ??? ... although I like MT's comment also which avoids this.

impact the amount of ACK traffic sent. Reducing the rate at which
acknowledgments are generated in constrained situations can preserve resources
and improve overall performance, for both network processing as well as

> application-relevant metrics.

Reviewer 2: I can't suggest over the diff in the way I'd like, but here is my take on this:

```md
# Acknowledgment Efficiency

QUIC version 1 without extensions uses an acknowledgment strategy based on TCP.
That is, every other packet is acknowledged.  However, generating and processing
QUIC acknowledgments can consume significant resources, both in terms of link
utilization and processing time.  Some applications might be able to improve
overall performance by using alternative strategies that reduce the rate of
acknowledgments.
```

This drops the "thinning" text (that's for the other doc).

There is a lot more to unpack here, but the transport doc already talks about this topic in more precise terms.

Reviewer: That looks good, OK also to me.

# ACK-only packets on constrained links

The stragegy on sending acknowlegdements in QUIC version 1 is adopted from
TCP, meaning that usually an QUIC implementation is expected to send one

> packet containing an ACK frame for each two packets received. 

Author: ```suggestion
packet containing an ACK frame for each two packets received.
```

The cost of sending acknowledgments - in processing cost or link
utilization - could be a significant proportion of available resources if

> these resources are constrained. With TCP network based mechanisim exists
> that perform so called ACK thining in such situations. As "only-ACK" packets
> are not reliably detecable in QUIC and drop of those packets can have larger
> performance impcats than with TCP, network based ACK thining is not suitable
> for QUIC traffic. Instead QUIC endpoint need to control the ACK rate

Reviewer 3: ```suggestion
these resources are constrained.  "ACK-only" packets are not reliably
detectable in QUIC, so network ACK thinning done for TCP is not suitable
for QUIC traffic. Instead QUIC endpoint need to control the ACK rate
```

appropriately and some implenentation may even provide an interface to

> impact the amount of ACK traffic sent. Reducing the rate at which
> acknowledgments are generated can preserve constraint resources and
> in such a situation improve overall performance, for both network processing
> as well as application-relevant metrics.

Reviewer 3: ```suggestion
impact the amount of ACK traffic sent. Reducing the rate at which
acknowledgments are generated in constrained situations can preserve resources
and improve overall performance, for both network processing as well as
application-relevant metrics.
```

packet containing an ACK frame for each two packets received.
The cost of sending acknowledgments - in processing cost or link
utilization - could be a significant proportion of available resources if
these resources are constrained.  "ACK-only" packets are not reliably
detectable in QUIC, so network ACK thinning done for TCP is not suitable
for QUIC traffic. Instead QUIC endpoint need to control the ACK rate

> appropriately and some implenentation may even provide an interface to

Reviewer 3: ```suggestion
for QUIC traffic. Instead QUIC endpoints need to control the ACK rate
appropriately and some implenentations may even provide an interface to
```
